frontend/src/routes/groups/[group_id]/+page.svelte

```diff
@@ -26,8 +43,129 @@
 			title: 'Trail Maintenance'
 		}
 	];
+
+	const libre311 = useLibre311Service();
+
+	async function handleAddNewService() {
+		newServiceName = stringValidator(newServiceName);
+		newServiceCode = stringValidator(newServiceCode);
+		newServiceDescription = stringValidator(newServiceDescription);
+		newServiceDefinition = stringValidator(newServiceDefinition);
+		newServiceGroupId = numberValidator(newServiceGroupId);
+
+		const resultSet = new Set([
+			newServiceName.type,
+			newServiceCode.type,
+			newServiceDescription.type,
+			newServiceDefinition.type,
+			newServiceGroupId.type
+		]);
+		if (resultSet.has('invalid')) return;
+
+		console.log(
+			await libre311.createService({
+				service_code: newServiceCode.value ?? '',
+				service_name: newServiceName.value ?? '',
+				description: newServiceDescription.value ?? '',
+				service_definition: newServiceDefinition.value ?? '',
+				group_id: newServiceGroupId.value ?? -1
```

Comment:
Are all of the nullish coalesces needed?  If they are showing up as `optional` it could be a problem of not doing the type narrowing correctly.